client/internal/wssender.go

```diff
@@ -62,10 +73,43 @@ func (s *WSSender) StoppingErr() error {
 	return s.err
 }
 
+// SetHeartbeatInterval sets the heartbeat interval and triggers timer reset.
+func (s *WSSender) SetHeartbeatInterval(d time.Duration) {
+	s.heartbeatIntervalSeconds.Store(int64(d.Seconds()))
+	select {
+	case s.heartbeatIntervalUpdated <- struct{}{}:
+	default:
+	}
+}
+
+func (s *WSSender) shouldSendHeartbeat() <-chan time.Time {
+	t := s.heartbeatTimer
+
+	if !t.Stop() {
+		select {
+		case <-t.C:
+		default:
+		}
+	}
```

Comment:
Please add a comment to explain what this does.

Reply:
Added.